### PR TITLE
LRDOCS-5527 Portal startup phases

### PR DESCRIPTION
**Pull request by jhinkey**
Hi Ray,

The startup phases reference article is ready for review. Thanks!

https://issues.liferay.com/browse/LRDOCS-5527

### CODE REVIEW

**Comment by jhinkey** [comment]
Hi Ray,
When might you be able to review the Startup Phases article? I based it on our discussion so it shouldn't need too much work. Thanks.

File: develop/reference/articles/03-back-end/01-portal-startup-phases.markdown [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+events they fire. 
+
+## Acting on Events [](id=acting-on-events)
+
+The ways to act on events depends on the event type. These subsections describe
+the event types. 
+
+### ModuleServiceLifecycle Events [](id=moduleservicelifecycle-events)
+
+[You can wait for and act on `ModuleServiceLifecycle` events.](/develop/tutorials/-/knowledge_base/7-1/waiting-on-lifecycle-events)
```
Comment (rotty3000) [inline develop/reference/articles/03-back-end/01-portal-startup-phases.markdown]: ok... maybe we don't need those inline descriptions... I'll let you decide on that.

File: develop/reference/articles/03-back-end/01-portal-startup-phases.markdown [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+Here's the phase's activity sequence:
+
+1.  Set up low level utilities such as logging and those in
+    [`PortalUtil`](@platform-ref@/7.1-latest/javadocs/portal-kernel/com/liferay/portal/kernel/util/PortalUtil.html)
+    and 
+    [`InitUtil`](@platform-ref@/7.1-latest/javadocs/portal-impl/com/liferay/portal/util/InitUtil.html). 
+
+2.  OSGi framework is initialized. 
+
+3.  Spring Phase 1: INFRASTRUCTURE beans are loaded.
```
[the remark this answers is not in the document]
Comment (jhinkey) [reply develop/reference/articles/03-back-end/01-portal-startup-phases.markdown]: @rotty3000 Do all of the beans come from the context files specified in the `spring.infrastrustructure.configs` property?

File: develop/reference/articles/03-back-end/01-portal-startup-phases.markdown [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    [OSGi services](/develop/tutorials/-/knowledge_base/7-1/osgi-services-and-dependency-injection-with-declarative-services). 
+
+5.  OSGi framework starts. 
+
+    1.  Static bundles are installed and started. 
+    2.  Dynamic bundles are started. 
+
+6.  OSGi framework starts the runtime.  
+
+7.  Spring Phase 2: MAIN Spring beans are loaded. Triggers the
```
[the remark this answers is not in the document]
Comment (jhinkey) [reply develop/reference/articles/03-back-end/01-portal-startup-phases.markdown]: @rotty3000 Do all of the MAIN Spring beans come from the context files specified in the property `spring.configs` property?

File: develop/reference/articles/03-back-end/01-portal-startup-phases.markdown (hunk 2 above)
[the remark this answers is not in the document]
Comment (rotty3000) [reply develop/reference/articles/03-back-end/01-portal-startup-phases.markdown]: yes

File: develop/reference/articles/03-back-end/01-portal-startup-phases.markdown (hunk 3 above)
[the remark this answers is not in the document]
Comment (rotty3000) [reply develop/reference/articles/03-back-end/01-portal-startup-phases.markdown]: yes

**Comment by jhinkey** [comment]
Thanks. Sent https://github.com/sez11a/liferay-docs/pull/3527

**Pull request merged**